crates/sui-indexer-alt-jsonrpc/src/lib.rs

```diff
@@ -76,6 +84,7 @@ impl RpcService {
         let RpcArgs {
             rpc_listen_address,
             max_rpc_connections,
+            ..
```

Comment:
This is a sign that `fullnode_rpc_url` doesn't belong on this struct -- imagine that this module is similar to `Indexer` and `IndexerArgs` (part of an RPC service framework).

Can you create a separate clap `Args` struct for the `WriteArgs`, in the `write` module, instead? Then you can `flatten` that struct into the `rpc` sub-command's list of fields. That way you can keep `RpcArgs` as just the flags passed to initialise the core of the RPC service, but still accept the fullnode URL as a command-line argument.

Reply:
Makes sense. I would indeed consider this similar to, for example, db url, which is a flattened arg.